coordinator/coordinator.go

```diff
@@ -584,14 +585,26 @@ func getOrderedSourceMappings(sources []metadata.NameVariant, sourceMap map[stri
 func (c *Coordinator) runPrimaryTableJob(source *metadata.SourceVariant, resID metadata.ResourceID, offlineStore provider.OfflineStore, schedule string) error {
 	c.Logger.Info("Running primary table job on resource: ", resID)
 	providerResourceID := provider.ResourceID{Name: resID.Name, Variant: resID.Variant, Type: provider.Primary}
-	if !source.IsPrimaryDataSQLTable() {
-		return fferr.NewInvalidArgumentError(fmt.Errorf("%s is not a primary table", source.Name()))
+
+	if !source.IsPrimary() {
+		return fferr.NewInvalidArgumentError(fmt.Errorf("%s is not a primary table", source.PrimaryDataLocationType()))
+	}
+
+	var primarySource metadata.PrimarySource
+	var srcErr error
+	switch source.PrimaryDataLocationType() {
+	case metadata.PrimaryDataSQLTable:
+		primarySource, srcErr = source.PrimaryDataSQLTable()
+	case metadata.PrimaryDataFilePath:
+		primarySource, srcErr = source.PrimaryDataFilePath()
+	default:
+		srcErr = fferr.NewInvalidArgumentError(fmt.Errorf("source location type not implemented: %s", source.PrimaryDataLocationType()))
```

Comment:
This is surely an internal error? Invalid Argument Type makes me think the user gave us a bad arg

Reply:
Also one of the reasons I don't love how we do fferr